### PR TITLE
[FEATURE] The "column.histogram" metric must support integer values of the "bins" parameter for all execution engine options.

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* The "column.histogram" metric current supports the integer values of the "bins" parameter only for `PandasExecutionEngine", whereas all `ExecutionEngine` options support the `list` format of the "bins" parameter.  However, supplying the number of "bins" as an integer is a proper default format (e.g., `bins = 10`) with the system's responsibility to compute the actual "bins" intervals and convert them to the `list` representation`.  The integer value of "bins" is the default `variable` in the `DataAssistant` application of the Rule-Based Profiler framework.  Now it is supported by all `ExecutionEngine` options.
* Unit tests for the "column.histogram" metric have been added.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-465/GREAT-982
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/metrics/column_aggregate_metrics/column_histogram.py [hunk 1]
```diff
@@ -69,10 +99,32 @@ def _sqlalchemy(
...
+                rows: List[Row] = sqlalchemy_engine.execute(query).fetchall()
+                row: Row
+                column_values: np.ndarray = np.asarray([row[0] for row in rows])
+                bins = np.histogram_bin_edges(column_values, bins=bins)
+                bins = bins.tolist()
+            except ProgrammingError as pe:
+                exception_message: str = "An SQL syntax Exception occurred."
+                exception_traceback: str = traceback.format_exc()
+                exception_message += f'{type(pe).__name__}: "{str(pe)}".  Traceback: "{exception_traceback}".'
+                logger.error(exception_message)
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/metrics/column_aggregate_metrics/column_histogram.py:126]: ❤️

File: tests/expectations/metrics/test_core.py [hunk 2]
```diff
@@ -524,6 +524,94 @@ def test_quantiles_metric_spark(spark_session):
...
+    )
+    results = engine.resolve_metrics(
+        metrics_to_resolve=(desired_metric,), metrics=metrics
+    )
+    metrics.update(results)
+    assert results == {desired_metric.id: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
+
+
+def test_column_histogram_metric_sa(sa):
+    engine = build_sa_engine(pd.DataFrame({"a": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]}), sa)
```
Comment (Shinnnyshinshin) [inline tests/expectations/metrics/test_core.py:556]: would it be worth putting typehints here?

File: tests/expectations/metrics/test_core.py [hunk 3]
```diff
@@ -524,6 +524,94 @@ def test_quantiles_metric_spark(spark_session):
     assert results == {desired_metric.id: [1.0, 2.0, 3.0]}
 
 
+def test_column_histogram_metric_pd():
+    engine = build_pandas_engine(pd.DataFrame({"a": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]}))
```
Comment (Shinnnyshinshin) [inline tests/expectations/metrics/test_core.py:528]: and here?

**Review by Shinnnyshinshin — APPROVED** [review]
Love it :) Just 2 small questions.

File: tests/expectations/metrics/test_core.py (hunk 2 above)
Comment (alexsherstinsky) [reply tests/expectations/metrics/test_core.py:556]: @Shinnnyshinshin We have not done it throughout this test module -- so perhaps as a global PR (a different one)?

File: tests/expectations/metrics/test_core.py (hunk 3 above)
Comment (alexsherstinsky) [reply tests/expectations/metrics/test_core.py:528]: @Shinnnyshinshin Same -- style -- we would do it throughout...

File: great_expectations/expectations/metrics/column_aggregate_metrics/column_histogram.py [hunk 4]
```diff
@@ -69,10 +99,32 @@ def _sqlalchemy(
...
+                .where(sa.column(column) != None)
+                .order_by(sa.column(column).asc())
+                .select_from(selectable)
+            )
+            try:
+                rows: List[Row] = sqlalchemy_engine.execute(query).fetchall()
+                row: Row
+                column_values: np.ndarray = np.asarray([row[0] for row in rows])
+                bins = np.histogram_bin_edges(column_values, bins=bins)
+                bins = bins.tolist()
```
Comment (anthonyburdi) [inline great_expectations/expectations/metrics/column_aggregate_metrics/column_histogram.py:121]: Is this going to work on larger data? Won't this load all data to the node running GE? Can we offload instead to the database i.e. put this computation in the query?

**Review by anthonyburdi — COMMENTED** [review]
See comment about running on larger data

**Pull request merged**